router/pkg/authentication/authentication.go

```diff
@@ -93,12 +93,13 @@ func Authenticate(ctx context.Context, authenticators []Authenticator, p Provide
 			joinedErrors = errors.Join(joinedErrors, err)
 			continue
 		}
-		if claims != nil {
-			return &authentication{
-				authenticator: auth.Name(),
-				claims:        claims,
-			}, nil
-		}
+
+		// If authentication succeeds, we return the authentication
```

Comment:
This was a flaw. As long as the authenticator return no error, we can assume the authentication was successful even when no claims have been configured.